--- libapreq2-2.17/glue/perl/xsbuilder/apreq_xs_tables.h.orig	2022-08-18 11:18:20.000000000 +0200
+++ libapreq2-2.17/glue/perl/xsbuilder/apreq_xs_tables.h	2025-07-28 11:50:53.698464188 +0200
@@ -42,7 +42,7 @@
  */
 
 static int apreq_xs_cookie_table_magic_copy(pTHX_ SV *sv, MAGIC *mg, SV *nsv,
-                                            const char *name, int namelen)
+                                            const char *name, I32 namelen)
 {
     /* Prefetch the value whenever the table iterator is > 0 */
     MAGIC *tie_magic = mg_find(nsv, PERL_MAGIC_tiedelem);
@@ -151,7 +151,7 @@
  */
 
 static int apreq_xs_param_table_magic_copy(pTHX_ SV *sv, MAGIC *mg, SV *nsv,
-                                  const char *name, int namelen)
+                                  const char *name, I32 namelen)
 {
     /* Prefetch the value whenever the table iterator is > 0 */
     MAGIC *tie_magic = mg_find(nsv, PERL_MAGIC_tiedelem);
